Automattic/jetpack
# Connection: verify REST API errors

#### Changes proposed in this Pull Request:
Currently only XMLRPC errors for requests coming from WPCOM get added to the "Verified Errors" list.
This PR also adds there the errors that occur if an REST API request to WPCOM fails.

#### Other information:


#### Jetpack product discussion
p9dueE-5zL-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Connect Jetpack.
2. Use the "Jetpack Debug -> Broken Token" plugin to set invalid blog and user tokens.
3. Go to Jetpack Dashboard to trigger the error.
4. Go to "Jetpack Debug - Connection Errors" and confirm an error with the code `unknown_token` shows up in both "Current Unverified Errors" and "Current Verified Errors" sections. The error should contain the `error_type: rest` value.
5. Click on the "Clear all errors" button, reload the page and confirm the errors have disappeared.
6. Run [Jetpack Debug Tool](https://jptools.wordpress.com/debug/) over the site, go back to the "Jetpack Debug -> Connection Errors".
7. Confirm that the error `no_valid_blog_token` was created with `error_type: xmlrpc`. If no errors were created, they might've been blocked by the "gate". Please remove the transients from the database and run the "Jetpack Debug Tool" again:
```
delete from wp_options where option_name like "%connection_error_reporting_gate%";
```
